test/src/test/java/org/corfudb/runtime/clients/LogUnitClientTest.java

```diff
@@ -400,4 +410,77 @@ public void CorruptedDataReadThrowsException() throws Exception {
                 .isInstanceOf(ExecutionException.class)
                 .hasCauseInstanceOf(DataCorruptionException.class);
     }
+
+    /**
+     * Testing that the clientId/ThreadId is persisted and that two
+     * LogData entries are equal if they have the same runtime and are coming
+     * from the same thread.
+     * @throws Exception
+     */
+    @Test
+    public void logDataWrittenIsEqualToLogDataRead() throws Exception {
+        long address = 0L;
+        LogData ld = getLogDataWithoutId(address);
+
+        UUID clientId = UUID.randomUUID();
+        ld.setId(clientId);
+        client.write(ld);
+
+        LogData ldPrime = client.read(address).get().getAddresses().get(address);
+
+        assertThat(ld).isEqualTo(ldPrime);
+        assertThat(ldPrime.getMetadataMap().get(IMetadata.LogUnitMetadataType.CLIENT_ID))
+                .isEqualTo(clientId);
+        assertThat(ldPrime.getMetadataMap().get(IMetadata.LogUnitMetadataType.THREAD_ID))
+                .isEqualTo(Thread.currentThread().getId());
+    }
+
+    /**
+     * Ensure that same LogData payload written by two different thread
+     * are not equals
+     * @throws Exception
+     */
+    @Test
+    public void logDataWrittenFromOtherThreadIsNotEqual() throws Exception {
+        long address = 0L;
+        UUID clientId = UUID.randomUUID();
+        LogData ldThisThread = getLogDataWithoutId(address);
+        ldThisThread.setId(clientId);
+        LogData ldOtherThread = getLogDataWithoutId(address);
+
+
+        Thread t = new Thread(()-> {
```

Comment:
There are build in methods t1() to use threads in the framework. Please use them.